eng/testing/tests.ioslike.targets

```diff
@@ -61,14 +62,21 @@
     </PropertyGroup>
 
     <ItemGroup>
-      <BundleFiles Condition="'%(AppleAssembliesToBundle._IsNative)' != 'true'"
-                   Include="@(AppleAssembliesToBundle)"         TargetDir="publish\%(AppleAssembliesToBundle.RecursiveDir)" />
-      <BundleFiles Include="@(AppleNativeFilesToBundle)"        TargetDir="publish\%(AppleNativeFilesToBundle.RecursiveDir)" />
-      <BundleFiles Include="$(RuntimeConfigFilePath)"           TargetDir="publish" />
+      <!-- 
+        Checks if the directory name "AppleAssembliesToBundle" is equal to the "AssemblyName",
+        if they match, consider it as the root directory and copy the files there,
+        otherwise, copy the files to a subdirectory.
+      -->
+      <BundleFiles Condition="'%(AppleAssembliesToBundle._IsNative)' != 'true' and ($([System.IO.Path]::GetFileName($([System.IO.Path]::GetDirectoryName('%(AppleAssembliesToBundle.Identity)')))) == '$(AssemblyName)' or '$(CustomMain)' != 'true')"
+              Include="@(AppleAssembliesToBundle)"                    TargetDir="publish" />
+      <BundleFiles Condition="'%(AppleAssembliesToBundle._IsNative)' != 'true' and $([System.IO.Path]::GetFileName($([System.IO.Path]::GetDirectoryName('%(AppleAssembliesToBundle.Identity)')))) != '$(AssemblyName)' and  '$(CustomMain)' == 'true'"
+                   Include="@(AppleAssembliesToBundle)"               TargetDir="publish\$([System.IO.Path]::GetFileName($([System.IO.Path]::GetDirectoryName('%(AppleAssembliesToBundle.Identity)'))))" />
```

Comment:
The same target is used for the library and runtime tests and the condition checks if the directory is equals to the assemblyName. In case of runtime tests that is true, but for library tests that creates recursive dirs. 

Probably better approach is to use more descriptive variable.